File: pyiArduinoI2Csht/cpp/NewAddress.cpp
// Данный пример меняет адрес модуля на указанный в argv[1].

#include <iostream>
#include "../iarduino_I2C_SHT.h"

iarduino_I2C_SHT sht;

uint8_t newAddress = 9;

int main(int argc, char** argv){
        if (argc > 1) {

                newAddress = atoi(argv[1]);

        }

        if (sht.begin()){

                std::cout << "На шине I2C найден модуль с адресом 0x";
                std::cout << std::hex << int(sht.getAddress());
                std::cout << " который является расширителем выводов\n";

                if (sht.changeAddress(newAddress)){

                        std::cout << "Адрес модуля изменён на 0x";
                        std::cout << std::hex << int(sht.getAddress()) << '\n';
                }
                else {

                        std::cout << "Адрес модуля изменить не удалось!" << '\n';

                }
        }
        else {

                std::cout << "Расширитель выводов не найден!" << '\n';

        }
}

//  ПРИМЕЧАНИЕ:
//  Данная программа демонстрирует не только возможность смены адреса
//  на указанный в переменной newAddress, но и обнаружение, и вывод
//  текущего адреса модуля на шине I2C.

File: pyiArduinoI2Csht/examples/newAddress.py
# ПРИМЕР СМЕНЫ АДРЕСА МОДУЛЯ:

# Подключаем библиотеку для работы с датчиком температуры и влажности
from pyiArduinoI2Csht import *
import sys

# Объявляем объект sht для работы с функциями и методами библиотеки pyiArduinoI2Csht.
# Если при объявлении объекта указать адрес, например, sht(0xBB),
# то пример будет работать с тем модулем, адрес которого был указан.
sht = pyiArduinoI2Csht(None, NO_BEGIN)

# Если сценарию не были переданы аргументы
if len(sys.argv) < 2:
    # Назначаем модулю адрес (0x07 < адрес < 0x7F).
    newAddress = 0x09

# Иначе
else:
    # Новый адрес - первый аргумент
    newAddress = int(sys.argv[1])

# Если датчик найден
if sht.begin():
    print("Найден датчик %#.2x" % sht.getAddress())

    # Если адрес удалось изменить
    if sht.changeAddress(newAddress):
            print("Адрес изменён на %#.2x" % sht.getAddress())

    else:
            print("Адрес не изменён!")

else:
    print("Датчик не найден!")
